server/app/interactors/syncs/create_sync.rb

```diff
@@ -5,6 +5,9 @@ class CreateSync
     include Interactor
 
     def call
+      source = context.workspace.connectors.find_by(id: context.sync_params[:source_id])
+      default_cursor_field = source&.catalog&.default_cursor_field(context.sync_params[:stream_name])
```

Comment:
Is there any instance that we don't have catalog? If not then  we should not be using `source&.catalog`